src/common/utils/types.h

```diff
@@ -19,8 +19,30 @@
 #include <variant>
 
 namespace raksha {
-  // Type alias for monostate that sounds less scary.
-  using Unit = std::monostate;
+// Type alias for monostate that sounds less scary.
+using Unit = std::monostate;
+
+// An implementation of a static, type-level, 'if'.
+// Useful for writing typesafe, code generically (especially for generic constness).
+// Based on SAT's laser (specifically 'misc_utils.h').
+template <bool Flag, typename T, typename F>
+struct SelectType {
+    using Result = T;
+};
+
+template <typename T, typename F>
+struct SelectType<true, T, F> {
+    using Result = T;
+};
+
+template <typename T, typename F>
+struct SelectType<false, T, F> {
+    using Result = F;
+};
+
+template <bool IsConst, typename T>
+using CopyConst = typename SelectType<IsConst, const T, T>::Result;
```

Comment:
I just realized, I think `SelectType` is the same as `std::conditional` (https://en.cppreference.com/w/cpp/types/conditional). That could reduce the size of this PR.

Reply:
Thanks Mark, will switch to using that and remove the associated new tests and types.

Reply:
Also, this could perhaps be more robust if `std::add_const` were used: https://en.cppreference.com/w/cpp/types/add_cv

Reply:
Another good find :) Thanks Mark. Seems the std lib has nice helpers `std::conditional_t` and `std::add_const_t` that make things a little easier.

Done! :)